application/src/main/data/json/system/widget_types/rpc_debug_terminal.json

```diff
@@ -11,7 +11,7 @@
                   var params;\n                            if (spaceIndex === -1) {\n                                spaceIndex = localCommand.length;\n                            }\n                            var name = localCommand.substr(0, spaceIndex);\n                            var args = localCommand.substr(spaceIndex + 1);\n                            if (args.length) {\n                                try {\n                                    params = JSON.parse(args);\n                                } catch (e) {\n                                    params = args;\n                                }\n                            }\n                            performRpc(this, name, params, requestUUID);\n                        }\n                    }\n                } catch(e) {\n                    this.error(new String(e));\n                }\n            } else {\n               this.echo('');\n            }\n        }, {\n            greetings: greetings,\n            prompt: prompt,\n            enabled: rpcEnabled,\n    });\n    \n    if (!rpcEnabled) {\n        terminal.error('No RPC target detected!').pause();\n    }\n    \n    $('#device-terminal').on('focusout.terminal', function(e) {\n            if (!$(e.relatedTarget).closest('#device-terminal').length && e.relatedTarget) {\n                terminal.blur();\n                \n            }    \n    });\n    \n    self.terminal = terminal;\n}\n\nfunction printUsage(terminal) {\n    var commandsListText = '\\n[[b;#fff;]Usage:]\\n';\n    commandsListText += '   <method> [params body]]\\n\\n';\n    commandsListText += '[[b;#fff;]Example 1:]\\n'; \n    commandsListText += '   myRemoteMethod1 myText\\n\\n'; \n    commandsListText += '[[b;#fff;]Example 2:]\\n'; \n    commandsListText += '   myOtherRemoteMethod \"{\\\\\\\"key1\\\\\\\": 2, \\\\\\\"key2\\\\\\\": \\\\\\\"myVal\\\\\\\"}\\\"\\n'; \n    terminal.echo(new String(commandsListText));\n}\n\nfunction performRpc(terminal, method, params, requestUUID) {\n    terminal.pause();\n    self.ctx.controlApi.sendTwoWayCommand(method, params, requestTimeout, requestPersistent, persistentPollingInterval, requestUUID).subscribe(\n        function success(responseBody) {\n            terminal.echo(JSON.stringify(responseBody));\n            terminal.echo(' ');\n            terminal.resume();\n        },\n        function fail() {\n            var errorText = self.ctx.defaultSubscription.rpcErrorText;\n            terminal.error(errorText);\n            terminal.echo(' ');\n            terminal.resume();\n        }\n    );\n}\n\nfunction uuidv4() {\n  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {\n    var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);\n    return v.toString(16);\n  });\n}\n\nself.onDestroy = function() {\n    $('#device-terminal').off('focusout.terminal');\n    if (self.terminal) {\n        self.terminal.destroy();\n    }\n    self.ctx.controlApi.completedCommand();\n}",
```

Comment:
`    $('#device-terminal').on('focusout.terminal', function(e) {
            if (!$(e.relatedTarget).closest('#device-terminal').length && e.relatedTarget) {
                terminal.blur();
                
            }    
    });`

Reply:
During the study of the problem, it was found that blur does not work correctly only if you click on another widget, and you need to understand why the event does not propagate.